torch/csrc/dynamo/eval_frame.c

```diff
@@ -476,6 +479,26 @@ static PyObject* call_guard_fail_hook(
   return result;
 }
 
+static PyObject* call_profiler_start_hook(PyObject* name_str) {
+  if (profiler_start_hook == NULL) return NULL;
+  if (name_str == NULL) return NULL;
+  PyObject* args = PyTuple_Pack(1, name_str);
+  if (args == NULL) return NULL;
+  PyObject* result = PyObject_CallObject(profiler_start_hook, args);
+  Py_DECREF(args);
+  return result;
+}
+
+static void call_profiler_end_hook(PyObject* record) {
+  // 'record' obj is the return value of calling _start_hook()
+  if (profiler_end_hook == NULL) return;
+  if (record == NULL) return;
+  PyObject* args = PyTuple_Pack(1, record);
+  if (args == NULL) return;
+  PyObject_CallObject(profiler_end_hook, args);
```

Comment:
Memory leak?  Return value is owned, but you don't decrement refcount.

Reply:
good catch, thanks!